pkgs/tools/graphics/directx-shader-compiler/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, fetchFromGitHub, cmake, python3, git }:
+
+stdenv.mkDerivation rec {
+  pname = "directx-shader-compiler";
+  version = "1.5.2010";
+
+  # Put headers in dev, there are lot of them which aren't necessary for
+  # using the compiler binary.
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "Microsoft";
+    repo = "DirectXShaderCompiler";
+    rev = "b640fa4ba4afb8eb9f15b4a34ab5187c7d94de97";
+    sha256 = "0ccfy1bfp0cm0pq63ri4yl1sr3fdn1a526bsnakg4bl6z4fwrnnj";
+    fetchSubmodules = true;
```

Comment:
Is it an option to use the nix versions of the submodules?

Reply:
possibly, although the versions use here are known-good.

Reply:
Do they track a specific version or some commit? If we run tests we could also use the system ones which is preferred but it is also fine if you don't want to risk breakage.

Reply:
GoogleTest is years old

none of them are tracking tags

Reply:
perhaps an issue for a future PR!

Reply:
I would prefer if you could do this in this PR.

Reply:
I'm not going to do that. It's potentially a whole bunch of yak-shaving for little benefit.